File: docs/snippets/all/archetypes/line_strips2d_segments_simple.py
"""Log a couple 2D line segments using 2D line strips."""

import numpy as np

import dalaran as dl
import dalaran.blueprint as dlb

dl.init("dalaran_example_line_segments2d", spawn=True)

dl.log(
    "segments",
    dl.LineStrips2D(np.array([[[0, 0], [2, 1]], [[4, -1], [6, 0]]])),
)

# Set view bounds:
dl.send_blueprint(
    dlb.Spatial2DView(
        visual_bounds=dlb.VisualBounds2D(x_range=[-1, 7], y_range=[-3, 3])
    )
)

File: docs/snippets/all/archetypes/line_strips2d_segments_simple.cpp
// Log a couple 2D line segments using 2D line strips.

#include <dalaran.hpp>

int main(int argc, char* argv[]) {
    const auto rec = dalaran::RecordingStream("dalaran_example_line_segments2d");
    rec.spawn().exit_on_failure();

    std::vector<std::vector<std::array<float, 2>>> points = {
        {{0.f, 0.f}, {2.f, 1.f}},
        {{4.f, -1.f}, {6.f, 0.f}},
    };
    rec.log("segments", dalaran::LineStrips2D(points));

    // TODO(#5520): log VisualBounds2D
}
